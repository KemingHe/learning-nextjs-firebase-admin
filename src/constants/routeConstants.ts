export const HOME_ROUTE: string = "/";

// Firebase client auth routes.
export const FB_CLIENT_HOME_ROUTE: string = "/firebase-client-auth";
export const FB_CLIENT_DASHBOARD_ROUTE: string =
  "/firebase-client-auth/dashboard";
export const FB_CLIENT_SIGN_IN_ROUTE: string = "/firebase-client-auth/signin";
export const FB_CLIENT_VERIFY_ROUTE: string = "/firebase-client-auth/verify";

// Firebase admin auth routes.
export const FB_ADMIN_HOME_ROUTE: string = "/firebase-admin-auth";
export const FB_ADMIN_DASHBOARD_ROUTE: string =
  "/firebase-admin-auth/dashboard";
export const FB_ADMIN_SIGN_IN_ROUTE: string = "/firebase-admin-auth/signin";

// Debug routes.
export const DEBUG_ROUTE: string = "/debug";
