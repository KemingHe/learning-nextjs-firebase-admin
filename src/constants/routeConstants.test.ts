import { describe, expect, it } from "vitest";

import * as routes from "./routeConstants";

describe("routeConstants", () => {
  it("exports the correct routes", () => {
    expect(routes.HOME_ROUTE).toBe("/");

    // Firebase client auth routes.
    expect(routes.FB_CLIENT_HOME_ROUTE).toBe("/firebase-client-auth");
    expect(routes.FB_CLIENT_DASHBOARD_ROUTE).toBe(
      "/firebase-client-auth/dashboard"
    );
    expect(routes.FB_CLIENT_SIGN_IN_ROUTE).toBe("/firebase-client-auth/signin");
    expect(routes.FB_CLIENT_VERIFY_ROUTE).toBe("/firebase-client-auth/verify");

    // Firebase admin auth routes.
    expect(routes.FB_ADMIN_HOME_ROUTE).toBe("/firebase-admin-auth");
    expect(routes.FB_ADMIN_DASHBOARD_ROUTE).toBe(
      "/firebase-admin-auth/dashboard"
    );
    expect(routes.FB_ADMIN_SIGN_IN_ROUTE).toBe("/firebase-admin-auth/signin");

    // Debug routes.
    expect(routes.DEBUG_ROUTE).toBe("/debug");
  });

  it("matches the snapshot", () => {
    expect(routes).toMatchSnapshot();
  });
});
