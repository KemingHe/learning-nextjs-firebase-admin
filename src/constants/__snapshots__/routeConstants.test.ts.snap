// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`routeConstants > matches the snapshot 1`] = `
{
  "DEBUG_ROUTE": "/debug",
  "FB_ADMIN_DASHBOARD_ROUTE": "/firebase-admin-auth/dashboard",
  "FB_ADMIN_HOME_ROUTE": "/firebase-admin-auth",
  "FB_ADMIN_SIGN_IN_ROUTE": "/firebase-admin-auth/signin",
  "FB_CLIENT_DASHBOARD_ROUTE": "/firebase-client-auth/dashboard",
  "FB_CLIENT_HOME_ROUTE": "/firebase-client-auth",
  "FB_CLIENT_SIGN_IN_ROUTE": "/firebase-client-auth/signin",
  "FB_CLIENT_VERIFY_ROUTE": "/firebase-client-auth/verify",
  "HOME_ROUTE": "/",
}
`;
